crypto/ec/ec_mult.c

```diff
@@ -77,10 +77,13 @@ static EC_PRE_COMP *ec_pre_comp_new(const EC_GROUP *group)
 
 EC_PRE_COMP *EC_ec_pre_comp_dup(EC_PRE_COMP *pre)
 {
-    int i;
-    if (pre != NULL)
-        CRYPTO_UP_REF(&pre->references, &i, pre->lock);
-    return pre;
+    int ref = 0;
+
+    if (pre != NULL) {
+        if (!CRYPTO_UP_REF(&pre->references, &ref, pre->lock))
+            return NULL;
+    }
+    return ref > 0 ? pre : NULL;
```

Comment:
Why not just return pre? If you get here and pre != NULL, the up_ref call was successful, ref was updated, and so ref should actually be > 1.